store/environment.go

```diff
@@ -139,7 +139,7 @@ func (s *Store) GetEnvironmentByID(ctx context.Context, id int) (*api.Environmen
 		return nil, errors.Wrapf(err, "failed to get environment with ID %d", id)
 	}
 	if envRaw == nil {
-		return nil, nil
+		return nil, common.Errorf(common.NotFound, "environment %d not found", id)
```

Comment:
This looks weird because `GetEnvironmentByID` doesn't guarantee finding one and other `GetXxx` functions in the store package all `return nil, nil` if `raw == nil`.

I think it's the caller's responsibility to check if the return value is nil.

Reply:
Let's fix it for now, but yes, the caller should handle the nil value - meaning not found.